internal/pkg/describe/service.go

```diff
@@ -406,3 +423,151 @@ func (e containerEnvVars) humanString(w io.Writer) {
 
 	printTable(w, headers, rows)
 }
+
+type secret struct {
+	Name        string `json:"name"`
+	Container   string `json:"container"`
+	Environment string `json:"environment"`
+	ValueFrom   string `json:"valueFrom"`
+}
+
+type secrets []*secret
+
+func (s secrets) humanString(w io.Writer) {
+	headers := []string{"Name", "Container", "Environment", "Value From"}
+	fmt.Fprintf(w, "  %s\n", strings.Join(headers, "\t"))
+	fmt.Fprintf(w, "  %s\n", strings.Join(underline(headers), "\t"))
+	sort.SliceStable(s, func(i, j int) bool { return s[i].Environment < s[j].Environment })
+	sort.SliceStable(s, func(i, j int) bool { return s[i].Container < s[j].Container })
+	sort.SliceStable(s, func(i, j int) bool { return s[i].Name < s[j].Name })
+	if len(s) > 0 {
+		valueFrom := s[0].ValueFrom
+		if _, err := arn.Parse(s[0].ValueFrom); err != nil {
+			// If the valueFrom is not an ARN, preface it with "parameter/"
+			valueFrom = fmt.Sprintf("parameter/%s", s[0].ValueFrom)
+		}
+		fmt.Fprintf(w, "  %s\n", strings.Join([]string{s[0].Name, s[0].Container, s[0].Environment, valueFrom}, "\t"))
+	}
+	for prev, cur := 0, 1; cur < len(s); prev, cur = prev+1, cur+1 {
+		valueFrom := s[cur].ValueFrom
+		if _, err := arn.Parse(s[cur].ValueFrom); err != nil {
+			// If the valueFrom is not an ARN, preface it with "parameter/"
+			valueFrom = fmt.Sprintf("parameter/%s", s[cur].ValueFrom)
+		}
+		cols := []string{s[cur].Name, s[cur].Container, s[cur].Environment, valueFrom}
+		if s[prev].Name == s[cur].Name {
+			cols[0] = dittoSymbol
+		}
+		if s[prev].Container == s[cur].Container {
+			cols[1] = dittoSymbol
+		}
+		if s[prev].Environment == s[cur].Environment {
+			cols[2] = dittoSymbol
+		}
+		if s[prev].ValueFrom == s[cur].ValueFrom {
+			cols[3] = dittoSymbol
+		}
+		fmt.Fprintf(w, "  %s\n", strings.Join(cols, "\t"))
+	}
+}
+
+func underline(headings []string) []string {
+	var lines []string
+	for _, heading := range headings {
+		line := strings.Repeat("-", len(heading))
+		lines = append(lines, line)
+	}
+	return lines
+}
+
+type internalEndpoint struct {
```

Comment:
nit: what do you think of moving the struct inside of the `marshalJSON` since it's the only place where it's used?